internal/resources/providers/awslib/lambda/event_source_mapping.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package lambda
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/lambda/types"
+
+	"github.com/elastic/cloudbeat/internal/resources/fetching"
+	"github.com/elastic/cloudbeat/internal/resources/utils/pointers"
+)
+
+type EventSourceMappingInfo struct {
+	EventSourceMapping types.EventSourceMappingConfiguration `json:"event_source_mapping_configuration"`
+	awsAccount         string
+	region             string
+}
+
+func (v EventSourceMappingInfo) GetResourceArn() string {
+	id := pointers.Deref(v.EventSourceMapping.UUID)
```

Comment:
Interesting the `UUID` contains an arn

Reply:
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/lambda@v1.54.6/types#EventSourceMappingConfiguration couldn't find any definition say it's or is not an ARN. Could you just confirm it? If it contains indeed ARN, I blame aws on bad naming :D

Reply:
It might not have a true ARN at all. I'll check.

Reply:
Ok, so it doesn't have an ARN per se.
```
aws --profile=cloud-security --region=us-east-1 lambda create-event-source-mapping --event-source arn:aws:sqs:us-east-1:378890115541:kuba-test.fifo --function-name kuba-test-func
{
    "UUID": "e4ee5f1b-53a3-4861-a892-d4153ac1e23b",
    "BatchSize": 10,
    "MaximumBatchingWindowInSeconds": 0,
    "EventSourceArn": "arn:aws:sqs:us-east-1:<ACCOUNT_ID>:kuba-test.fifo",
    "FunctionArn": "arn:aws:lambda:us-east-1:<ACCOUNT_ID>:function:kuba-test-func",
    "LastModified": "2024-06-17T13:21:47.009000+02:00",
    "State": "Creating",
    "StateTransitionReason": "USER_INITIATED",
    "FunctionResponseTypes": []
}
```

Just UUID. I think I'll make it return an empty ARN field, WDYT @romulets?

Reply:
If we don't have an ARN available, I'd rather not return it. Avoids unintended usage and bugs